[Tan Wei Jie] iP

tanweijie123 opened the pull request.

Commit:
Add Gradle support

tanweijie123 committed:
Level-1

tanweijie123 committed:
Level-2

tanweijie123 committed:
Level-3

tanweijie123 committed:
Level-4

tanweijie123 committed:
A-TextUiTesting
Edited runtest to include src directory and pause at the end
changed input and expected tc

tanweijie123 committed:
Level-5
Changed all exceptions to throw DukeException
Added Exceptions for Done, ToDo, deadline, event, and invalid command.

tanweijie123 committed:
Level-6
Added delete command; minor changes to exception output, changed println to accept varargs

tanweijie123 committed:
Changed all methods with new String[] variable to make use of varargs println.

tanweijie123 committed:
A-Enums
Added enums for command list.

tanweijie123 committed:
Include save function

Git will ignore data/duke.txt file

tanweijie123 committed:
Change date from using String to LocalDateTime

tanweijie123 committed:
Merge branch 'branch-Level-7'

tanweijie123 committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Duke.java

tanweijie123 committed:
Refactor Duke to make codes more OOP

tanweijie123 committed:
Organise the OOP class into package

duke.task contain all the different task creatable by user
duke.command contain all the different command usable
duke.util contain essential classes for Duke to run.
duke.exception contain exceptions for Duke programme

tanweijie123 committed:
Add JUnit test for Task, Deadline, Event

tanweijie123 committed:
Create a release v0.1

tanweijie123 committed:
Follow the Java Coding Standard

Switch statements are changed to case without indent.
If statement with no braces are added
import * changed to list inidividual class

tanweijie123 committed:
Add Javadocs for all public classes, methods, attributes.

tanweijie123 committed:
Add find feature to search all task description and display result based on search keyword.

tanweijie123 committed:
Merge branch 'branch-A-JavaDoc'

tanweijie123 committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/Duke.java

tanweijie123 committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/duke/util/Parser.java

tanweijie123 committed:
Add Javadoc for find feature

tanweijie123 committed:
Update typo in javadoc code

tanweijie123 committed:
Merge branch 'add-gradle-support'

tanweijie123 committed:
Fix compile error by modify unicode to \u format

tanweijie123 committed:
Add Checkstyle files checkstyle.xml suppression.xml

tanweijie123 committed:
Add checkstyle file. Update build.gradle checkstyle version

shaokiat commented on src/main/java/duke/util/TaskList.java:
```diff
@@ -0,0 +1,87 @@
+     *
+     * @return List of task outputs.
+     */
+    public List<String> export() {
+        return taskList
+                .stream()
+                .map(x -> x.getSaveToFileString())
+                .collect(Collectors.toList());
+    }
```
Good use of stream!

shaokiat commented on src/main/java/duke/util/Ui.java:
```diff
@@ -0,0 +1,160 @@
+     *
+     * @param message message to be output.
+     */
+    private static void println(String... message) {
```
Good use of varags

shaokiat commented on src/main/java/duke/util/Util.java:
```diff
@@ -0,0 +1,78 @@
+    public static LocalDateTime convertStringToDateTime(String dateTime) {
+        //Allow format of "YYYY-MM-dd HHmm", "dd/MM/yyyy HHmm"; Set HHmm to 0000 if not found.
+
+        try {
+            if (!dateTime.contains(" ")) {
+                dateTime = dateTime + " 0000"; //pad with time if the input only contains date.
+            }
+
+            if (dateTime.charAt(4) == '-') {
+                return LocalDateTime.parse(dateTime, DateTimeFormatter.ofPattern("yyyy-MM-dd HHmm"));
+            } else if (dateTime.charAt(2) == '/') {
+                return LocalDateTime.parse(dateTime, DateTimeFormatter.ofPattern("dd/MM/yyyy HHmm"));
+            } else {
+                throw new DukeException("Invalid Date / time format...");
```
Maybe could try out SimpleDateFormat as alternative

shaokiat reviewed (commented):
Well written and documented code! Good job. I was able to pick up good learning points from your code.

tanweijie123 commented:
> Well written and documented code! Good job. I was able to pick up good learning points from your code.

Hello. Thanks for the review and the kind words. Is there any coding standard violation that I had missed?

tanweijie123 committed:
Add JavaFX, FXML. Just finish Tutorial 4.

This commit save as a checkpoint before cleaning up Duke class

tanweijie123 committed:
Clean Duke class to remove javafx elements; Add title to main window

tanweijie123 committed:
Merge branch 'branch-Level-10'

tanweijie123 committed:
Integrate Duke into GUI output.

yan-soon commented on src/main/java/duke/command/AddCommand.java:
```diff
@@ -0,0 +1,35 @@
+ * {@code AddCommand} is a child of {@code Command} object.
+ *      On execution, it will add a given task into the task list.
+ */
+public class AddCommand extends Command {
+
+    private Task task;
+
+    /**
+     * Constructs a AddCommand which adds a new task to the list on execution.
+     *
+     * @param t The task to be added into the list of tasks.
```
I like your extraction of command classes, as it makes the code a lot cleaner and more comprehensive.

zeying99 replied:
same i agree :)

yan-soon commented on src/main/java/duke/util/Parser.java:
```diff
@@ -0,0 +1,117 @@
+            break;
+        case done:
+            try {
+                int selected = Integer.parseInt(cmd.substring(4).strip());
```
Perhaps the use of substrings and the significance of the numbers can be illustrated more clearly?

yan-soon commented on src/main/java/duke/util/Storage.java:
```diff
@@ -0,0 +1,95 @@
+        try {
+            Scanner readFile = new Scanner(file);
+            while(readFile.hasNextLine()) {
+                String ln = readFile.nextLine();
```
Perhaps there can be another javadoc here as well?

yan-soon commented on src/test/java/duke/task/ToDoTest.java:
```diff
@@ -0,0 +1,37 @@
+    @Test
+    public void todoCreation_normalInput_success() {
+        ToDo todo = new ToDo("read book");
+        Assertions.assertEquals("[T][\u2718] read book", todo.toString());
```
I like your thorough testing!

yan-soon reviewed (commented):
Overall, very clean and concise code! A lot of methods have been abstracted out which makes for easier comprehension on the user end.

zeying99 commented on src/main/java/duke/command/CommandList.java:
```diff
@@ -0,0 +1,16 @@
+/**
+ * The list of commands available for {@code Duke}.
+ */
+public enum CommandList {
```
great use of enum. Very clear of the possible tasks

zeying99 commented on src/main/java/duke/command/DeleteCommand.java:
```diff
@@ -0,0 +1,40 @@
+ */
+public class DeleteCommand extends Command {
+
+    private int idx;
```
perhaps you want to use the name index instead of idx to improve comprehensiveness

zeying99 commented on src/main/java/duke/task/Event.java:
```diff
@@ -0,0 +1,46 @@
+    }
+
+    /**
+     * {@inheritDoc}
```
Wow i learn something new today that javadoc can inherit too. Thank you

zeying99 reviewed (commented):
Great structure and nice javadocs. I have learnt a lot from your codes. Thank you. 👍

tanweijie123 committed:
Update build.gradle to point to main start up class

tanweijie123 committed:
Remove dead codes, magic numbers in Parser, Add throws for exceptions, Get static task deliminter from 1 source.

Removed dead codes from Duke.java. The old run method is not useful for GUI.
Parser to change the substring numbers to commandString and contentString, extract the multiple use of strip.
Added throws DukeException for those method that does not have it in the method signature.
Task_Delimiter will get from Util class, standardize for the whole programme. (single point to change delimiter)

tanweijie123 committed:
Add assertions for method pre-condition and checking

tanweijie123 committed:
Update todo

tanweijie123 committed:
Merge pull request #1 from tanweijie123/branch-A-Assertions

Add assertions for method pre-condition and checking

tanweijie123 committed:
Merge branch 'master' of https://github.com/tanweijie123/ip

tanweijie123 committed:
Merge branch 'master' into branch-A-CodeQuality

# Conflicts:
#	src/main/java/duke/task/Deadline.java
#	src/main/java/duke/task/Event.java
#	src/main/java/duke/task/ToDo.java
#	src/main/java/duke/util/Storage.java

tanweijie123 committed:
Merge pull request #2 from tanweijie123/branch-A-CodeQuality

PR branch-A-CodeQuality

tanweijie123 committed:
Add priority to tasks.

Tasks now have priority label.
List can be display based on priority level and reverse order.

tanweijie123 committed:
Add new / Modify existing test case to reflect the changes in priority type; Change GUI from 600x400 to 600x600

tanweijie123 committed:
Convert the entire command switch-case into function pointers

tanweijie123 committed:
Refactor Parse class to ensure SLAP

tanweijie123 committed:
Update Test Cases for priority related task.

Additional feature to extract priority from description string.

tanweijie123 committed:
Update GUI of Duke & minor comestic changes

tanweijie123 committed:
Add brief UG and Ui display

tanweijie123 committed:
Remove file tracking for build folder

tanweijie123 committed:
Update type
